app/models/one_signal_player.rb

```diff
@@ -24,4 +24,26 @@ class OneSignalPlayer < ApplicationRecord
   belongs_to :user
 
   enum platform: %i[web mobile]
+
+  def self.filter_player_ids(player_ids, notif_type)
+    filtered_player_ids = []
+
+    player_objects = OneSignalPlayer.includes(
+      user: :notification_settings
+    ).where(
+      id: player_ids
+    )
+
+    player_objects.each do |player|
+      user_setting = player.user.notification_settings.where(
```

Comment:
I don't think this will use the `includes` data because you included a `where` clause, so this is probably an N+1 query